### PR TITLE
feat: add type annotations in jina executors

### PR DESCRIPTION
**Pull request by rameshwara**
feat: add type annotations in jina executors

### CODE REVIEW

**Comment by rameshwara** [comment]
Due to the misplacement of the branch head, the commits made for adding annotations in drivers package were also included here. But all of them have now been reverted and only the type annotations for the executors are included in this pull request.

File: jina/executors/compound.py [hunk 1]
```diff
@@ -210,12 +210,12 @@ def is_updated(self) -> bool:
         return (self.components and any(c.is_updated for c in self.components)) or self._is_updated
 
     @is_updated.setter
-    def is_updated(self, val: bool):
+    def is_updated(self, val: bool) -> bool:
```
Comment (JoanFM) [inline jina/executors/compound.py:213]: this is not true, it is a setter so it does not have any return value, it returns `None`

File: jina/executors/compound.py [hunk 2]
```diff
@@ -210,12 +210,12 @@ def is_updated(self) -> bool:
...
         """Set :attr:`is_updated` for this :class:`CompoundExecutor`. Note, not to all its components """
         self._is_updated = val
 
     @is_trained.setter
-    def is_trained(self, val: bool):
+    def is_trained(self, val: bool) -> bool:
```
Comment (JoanFM) [inline jina/executors/compound.py:218]: same here

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/executors/compound.py (hunk 1 above)
Comment (rameshwara) [reply jina/executors/compound.py]: Done

File: jina/executors/compound.py (hunk 2 above)
Comment (rameshwara) [reply jina/executors/compound.py]: Done

**Review by JoanFM — APPROVED** [review]

**Pull request merged**